haystack/components/retrievers/auto_merging_retriever.py

```diff
@@ -38,24 +38,26 @@ class AutoMergingRetriever:
     from haystack.components.preprocessors import HierarchicalDocumentSplitter
     from haystack.components.retrievers.auto_merging_retriever import AutoMergingRetriever
     from haystack.document_stores.in_memory import InMemoryDocumentStore
+    from haystack.document_stores.types import DuplicatePolicy
 
     # create a hierarchical document structure with 3 levels, where the parent document has 3 children
     text = "The sun rose early in the morning. It cast a warm glow over the trees. Birds began to sing."
     original_document = Document(content=text)
-    builder = HierarchicalDocumentSplitter(block_sizes=[10, 3], split_overlap=0, split_by="word")
+    builder = HierarchicalDocumentSplitter(block_sizes={10, 3}, split_overlap=0, split_by="word")
     docs = builder.run([original_document])["documents"]
 
     # store level-1 parent documents and initialize the retriever
     doc_store_parents = InMemoryDocumentStore()
-    for doc in docs["documents"]:
-        if doc.meta["children_ids"] and doc.meta["level"] == 1:
-            doc_store_parents.write_documents([doc])
+    for doc in docs + [original_document]:
+        if doc.meta["__children_ids"] and doc.meta["__level"] in [0,1]:  # store the root document and level 1 documents
+            doc_store_parents.write_documents([doc], policy=DuplicatePolicy.SKIP)
```

Comment:
I might be wrong, but my impression is that if we just do `for doc in docs` (not including the original document), we don't need `DuplicatePolicy.SKIP`.